File: swimming-contest-software/Server/src/main/java/network/SwimmingRacesConcurrentServer.java
package network;

import services.SwimmingRaceServices;

import java.net.Socket;

public class SwimmingRacesConcurrentServer extends AbstractConcurrentServer {

    private final SwimmingRaceServices services;

    public SwimmingRacesConcurrentServer(int port, SwimmingRaceServices swimmingRaceServices) {
        super(port);
        logger.info("Server name: SwimmingRacesServer; Server type: concurrent");
        this.services = swimmingRaceServices;
    }

    @Override
    protected Thread createWorker(Socket client) {
        logger.info("Creating worker to handle request...");
        SwimmingRacesClientWorker clientWorker = new SwimmingRacesClientWorker(services, client);
        return new Thread(clientWorker);
    }
}

File: Server/Network/SwimmingRacesConcurrentServer.cs
using System.Net.Sockets;
using System.Threading;
using Model.Services;

namespace Server.Network;

public class SwimmingRacesConcurrentServer : AbstractConcurrentServer
{
    private readonly ISwimmingRaceServices _services;

    public SwimmingRacesConcurrentServer(string host, int port, ISwimmingRaceServices services) : base(host, port)
    {
        Logger.Info("Server name: SwimmingRacesServer; Server type: concurrent; Port: " + port);
        _services = services;
    }

    protected override Thread createWorker(TcpClient client)
    {
        Logger.Info("Creating worker to handle client...");
        var clientWorker = new SwimmingRacesClientWorker(_services, client);
        return new Thread(clientWorker.Run);
    }
}